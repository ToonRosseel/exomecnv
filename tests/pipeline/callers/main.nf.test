nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_callers"
    tag "pipeline_annotate"

    test("pipeline - exomedepth (CRAM input)") {

        when {
            params {
                outdir = "${outputDir}"
                exomedepth = true
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("exomedepth_cram") }
            )
        }
    }
    test("pipeline - exomedepth (CRAM input) + annotate") {

        when {
            params {
                outdir = "${outputDir}"
                exomedepth = true
                annotate = true
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("exomedepth_cram + annotate") }
            )
        }
    }
    test("pipeline - exomedepth (BAM input)") {

        when {
            params {
                input  = "${projectDir}/tests/inputs/samplesheet_bam.csv"
                outdir = "${outputDir}"
                exomedepth = true
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("exomedepth_bam") }
            )
        }
    }
}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}
