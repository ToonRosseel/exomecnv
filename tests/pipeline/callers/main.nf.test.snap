{
    "exomedepth_cram": {
        "content": [
            63,
            {
                "BCFTOOLS_SORT": {
                    "bcftools": 1.21
                },
                "BEDGOVCF": {
                    "bedgovcf": "0.1.0"
                },
                "BEDTOOLS_MAP": {
                    "bedtools": "2.31.1"
                },
                "CUSTOM_MERGECOUNTS": {
                    "paste": 9.3,
                    "mv": 9.3,
                    "cp": 9.3
                },
                "CUSTOM_REFORMATCOUNTS": {
                    "echo": "--version",
                    "cut": 9.3
                },
                "EXOMEDEPTH_CALL": {
                    "ExomeDepth": "1.1.18",
                    "R": "Rscript (R) version 4.4.3 (2025-02-28)"
                },
                "MOSDEPTH": {
                    "mosdepth": "0.3.10"
                },
                "Workflow": {
                    "nf-cmgg/exomecnv": "v2.0.0"
                }
            },
            [
                "exomedepth",
                "exomedepth/cnv_call",
                "exomedepth/cnv_call/Sample1.vcf.gz",
                "exomedepth/cnv_call/Sample1.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample10.vcf.gz",
                "exomedepth/cnv_call/Sample10.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample2.vcf.gz",
                "exomedepth/cnv_call/Sample2.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample3.vcf.gz",
                "exomedepth/cnv_call/Sample3.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample4.vcf.gz",
                "exomedepth/cnv_call/Sample4.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample5.vcf.gz",
                "exomedepth/cnv_call/Sample5.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample6.vcf.gz",
                "exomedepth/cnv_call/Sample6.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample7.vcf.gz",
                "exomedepth/cnv_call/Sample7.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample8.vcf.gz",
                "exomedepth/cnv_call/Sample8.vcf.gz.tbi",
                "exomedepth/cnv_call/Sample9.vcf.gz",
                "exomedepth/cnv_call/Sample9.vcf.gz.tbi",
                "exomedepth/counts",
                "exomedepth/counts/prep_F.merged.txt",
                "exomedepth/counts/prep_M.merged.txt",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_pipeline_software_mqc_versions.yml"
            ],
            [
                "Sample1.vcf.gz:md5,31b44a6ffaff46442b282b4b89f61035",
                "Sample1.vcf.gz.tbi:md5,47181a241dff14cf4affdeaff0fabcd0",
                "Sample10.vcf.gz:md5,7ea93e8494d1d04817ad76742a70dab5",
                "Sample10.vcf.gz.tbi:md5,4cb176febbc8c26d717a6c6e67b9c905",
                "Sample2.vcf.gz:md5,d1fba3a4f43f33ff84cbc9d63bdbb2e9",
                "Sample2.vcf.gz.tbi:md5,71fd3e30fcfac40f1fb8027577e46fa8",
                "Sample3.vcf.gz:md5,f2edf501c2d103cb9c8640562e402e9e",
                "Sample3.vcf.gz.tbi:md5,70174c214538e65ca6310c39eda72d9f",
                "Sample4.vcf.gz:md5,ec57e37cd698b511338cdf6c46aabc2f",
                "Sample4.vcf.gz.tbi:md5,7bc432502c79e36276d60d247d40604a",
                "Sample5.vcf.gz:md5,3169b1b93e6b82d6ac84dee24ad10f35",
                "Sample5.vcf.gz.tbi:md5,15572bdad0d2d5294ab403e6bf7ff3ba",
                "Sample6.vcf.gz:md5,e9bc56a2ec4b698f8ea7d2b3f0d85fc1",
                "Sample6.vcf.gz.tbi:md5,9dd93f302cabe245abd5d249a57c1866",
                "Sample7.vcf.gz:md5,a0b1980e9a4dde229d3a8fd2248931d2",
                "Sample7.vcf.gz.tbi:md5,6348efd14b2a65d9c92b2fe6292c7258",
                "Sample8.vcf.gz:md5,87fe426202d95494e87f15940d888e92",
                "Sample8.vcf.gz.tbi:md5,e2a8c69b8ff693ec525d85482bf76985",
                "Sample9.vcf.gz:md5,2074a5f25116dd0a581c84349bb97e7e",
                "Sample9.vcf.gz.tbi:md5,e51f4702088b4b17e012326c266410ad",
                "prep_F.merged.txt:md5,89a1a53ea465c6a07f47f82041d6fe19",
                "prep_M.merged.txt:md5,ade810cfa5a293c8142f02c54195ec04"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.0"
        },
        "timestamp": "2025-09-01T16:50:59.320838727"
    }
}